### PR TITLE
add createInferenceATF in BaseInferrableChecker

### PR DESCRIPTION
**Pull request by Jianchu**
Added a method createInferenceATF in BaseInferrableChecker, real checker overrides it and return a type system specific InferenceAnnotatedTypeFactory.

### CODE REVIEW

File: src/checkers/inference/InferenceAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -166,18 +166,8 @@ public CFTransfer createFlowTransferFunction(CFAbstractAnalysis<CFValue, CFStore
 
     @Override
     public TreeAnnotator createTreeAnnotator() {
-        final TreeAnnotator realInferenceTreeAnnotator;
-        if (realTypeFactory instanceof InferrableAnnotatedTypeFactory) {
```
Comment (wmdietl) [inline src/checkers/inference/InferenceAnnotatedTypeFactory.java]: You also need to remove InferrableAnnotatedTypeFactory.

File: src/checkers/inference/InferenceMain.java [hunk 2]
```diff
@@ -93,7 +93,7 @@
     private BaseAnnotatedTypeFactory realTypeFactory;
     private InferenceAnnotatedTypeFactory inferenceTypeFactory;
 
-    private final ConstraintManager constraintManager = new ConstraintManager();
+    private ConstraintManager constraintManager;
```
Comment (wmdietl) [inline src/checkers/inference/InferenceMain.java]: Is this change to the constraintManager related? Or just a general cleanup?

Comment (Jianchu) [inline src/checkers/inference/InferenceMain.java]: It's just a cleanup.

File: src/checkers/inference/InferenceAnnotatedTypeFactory.java (hunk 1 above)
Comment (Jianchu) [inline src/checkers/inference/InferenceAnnotatedTypeFactory.java]: InferrableAnnotatedTypeFactory is removed.

**Pull request merged**